uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestTest.php

```diff
@@ -153,91 +153,50 @@ public function testDeleteDeletesTheLeaveRequestItsBalanceChangesAndDates() {
-      'type_id' => 1,
-      'from_date' => CRM_Utils_Date::processDate('2016-01-04'),
-      'from_date_type' => $this->getLeaveRequestDayTypes()['All Day']['value'],
-      'to_date' => CRM_Utils_Date::processDate('2016-01-10'),
-      'to_date_type' => $this->getLeaveRequestDayTypes()['All Day']['value'],
-      'status_id' => $this->leaveRequestStatuses['More Information Requested']['id'],
-      'sequential' => 1,
-    ]);
+    $leaveRequest = LeaveRequestFabricator::fabricateWithoutValidation($this->getDefaultParams());
 
     $this->setExpectedException(
       'RuntimeException', "You can't change the Leave Request status from ".
-      $this->leaveRequestStatuses['More Information Requested']['id']. " to {$this->leaveRequestStatuses['Waiting Approval']['id']}"
+      $this->getDefaultParams()['status_id']. " to {$this->leaveRequestStatuses['Waiting Approval']['id']}"
     );
 
-    $this->getLeaveRequestServiceWhenStatusTransitionIsNotAllowed()->create([
+    $params = $this->getDefaultParams([
       'id' => $leaveRequest->id,
-      'type_id' => 1,
-      'contact_id' => $this->le,
-      'status_id' => $this->leaveRequestStatuses['Waiting Approval']['id'],
-      'from_date' => CRM_Utils_Date::processDate('2016-01-04'),
-      'from_date_type' => $this->getLeaveRequestDayTypes()['All Day']['value'],
-      'to_date' => CRM_Utils_Date::processDate('2016-01-10'),
-      'to_date_type' => $this->getLeaveRequestDayTypes()['All Day']['value'],
-    ], false);
+      'status_id' => $this->leaveRequestStatuses['Waiting Approval']['id']
+    ]);
+
+    $this->getLeaveRequestServiceWhenStatusTransitionIsNotAllowed()->create($params, false);
   }
 
   public function testCreateThrowsAnExceptionForLeaveContactWhenUpdatingLeaveStatusWithoutPermission() {
     //The leave manager creates a leave request with More information requested status
-    $leaveRequest = $this->getLeaveRequestServiceWhenCurrentUserIsLeaveManager()->create([
-      'type_id' => 1,
-      'contact_id' => $this->leaveContact,
-      'status_id' => $this->leaveRequestStatuses['More Information Requested']['id'],
-      'from_date' => CRM_Utils_Date::processDate('2016-01-04'),
-      'from_date_type' => $this->getLeaveRequestDayTypes()['All Day']['value'],
-      'to_date' => CRM_Utils_Date::processDate('2016-01-10'),
-      'to_date_type' => $this->getLeaveRequestDayTypes()['All Day']['value'],
-    ], false);
-
+    $params = $this->getDefaultParams(['status_id' => $this->leaveRequestStatuses['More Information Requested']['id']]);
+    $leaveRequest = $this->getLeaveRequestServiceWhenCurrentUserIsLeaveManager()->create($params, false);
 
     //The leave contact tries to change the status to 'Waiting Approval'
     //Even though it was a valid status transition but the leave contact does not have the permission
+    $params = $this->getDefaultParams([
+      'id' => $leaveRequest->id,
+      'status_id' => $this->leaveRequestStatuses['Waiting Approval']['id']
+    ]);
```

Comment:
you already have `$params`, so all you need to do is add the 'id' to id and change the status_id value